src/osd/scrubber/pg_scrubber.cc

```diff
@@ -2426,45 +2518,123 @@ ostream& PgScrubber::show(ostream& out) const
-			   std::string param,
-			   Formatter* f,
-			   stringstream& ss)
+int PgScrubber::asok_debug(
+  Formatter* f,
+  std::string_view prefix,
+  std::string_view cmd,
+  std::string_view param,
+  std::string_view arg1)
 {
-  dout(10) << __func__ << " cmd: " << cmd << " param: " << param << dendl;
+  dout(10)
+    << fmt::format(
+	 "asok_debug: pr={} cmd={}, param={}, val={}", prefix, cmd, param, arg1)
+    << dendl;
+
+  auto set_on = prefix == "scrubdebug" || prefix == "scrubdebug_set";
+  int ret{0};
 
-  if (cmd == "block") {
-    // 'm_debug_blockrange' causes the next 'select_range' to report a blocked
-    // object
+  // (code maintainer: see PR discussions re the various formatting options)
+  if (cmd == "block" && set_on) {
+    // cause the next 'select_range' to report a blocked object
     m_debug_blockrange = 10;  // >1, so that will trigger fast state reports
+    f->dump_string("op", "blocking scrub range");
+
+  } else if (cmd == "block" && !set_on) {
+    m_debug_blockrange = 0;
+    m_fsm->process_event(Unblocked{});
+    f->dump_string("op", "scrub range unblocked");
 
   } else if (cmd == "unblock") {
-    // send an 'unblock' event, as if a blocked range was freed
     m_debug_blockrange = 0;
     m_fsm->process_event(Unblocked{});
+    f->dump_string("op", "scrub range unblocked");
 
-  } else if ((cmd == "set") || (cmd == "unset")) {
-
-    if (param == "sessions") {
-      // set/reset the inclusion of the scrub sessions counter in 'query' output
-      m_publish_sessions = (cmd == "set");
-
-    } else if (param == "block") {
-      if (cmd == "set") {
-	// set a flag that will cause the next 'select_range' to report a
-	// blocked object
-	m_debug_blockrange = 10;  // >1, so that will trigger fast state reports
-      } else {
-	// send an 'unblock' event, as if a blocked range was freed
-	m_debug_blockrange = 0;
-	m_fsm->process_event(Unblocked{});
-      }
-    }
+  } else if (cmd == "sessions" && set_on) {
+    // set the inclusion of the scrub sessions counter in 'query' output
+    m_publish_sessions = true;
+    f->dump_string("op", "ok");
+
+  } else if (cmd == "sessions" && !set_on) {
+    // set the inclusion of the scrub sessions counter in 'query' output
+    m_publish_sessions = false;
+    f->dump_string("op", "ok");
+
+  } else if (cmd == "repdeny") {
+    debug_inject_reservation_err.response = ReservationErr::drop_reservation;
+    f->dump_string("op", "dropping next reservation request");
+
+  } else if (cmd == "repdelay") {
+    ret = dbg_reserv_delay(f, param, arg1);
+
+  } else {
+    ret = -EINVAL;
   }
+  return ret;
+}
 
+int PgScrubber::dbg_reserv_delay(
+    Formatter* f,
+    std::string_view param,
+    std::string_view value)
+{
+  milliseconds delay{4s};
+  if (!value.empty()) {
+    try {
+      delay = milliseconds{std::stol(std::string{value})};
+    } catch (const std::invalid_argument& e) {
+      f->dump_string("error", fmt::format("invalid delay value: {}", e.what()));
```

Comment:
`what()` returns a _pointer to a null-terminated string with explanatory information_. Does it make sense to print also the `value` that caused the exception?

Reply:
@rzarzynski : you are absolutely right. what() returns nothing useful ('stol').
Will fix.